### PR TITLE
feat(ci): replace set-output to GITHUB_OUTPUT

### LINKED ISSUE #22483: (ci): (set-output in Github Actions to be deprecated)
### Describe the feature

replace deprecated set-output command with GITHUB_OUTPUT

### Use Case

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
As per this URL, set-output of Github Actions is deprecated.

### Proposed Solution

`echo "::set-output name=dir::$(yarn cache dir)"`
replace to
`echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT`

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.45.0

### Environment details (OS name and version, etc.)

macOS 12.4

### PR DESCRIPTION
**Pull request by k1rnt**
Hi!
fixes #22483 

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by k1rnt** [comment]
Hi aws!

Replace deprecated features in Github Actions Commands.
❌ Feature must include changes to the README file.
❌ Feature must include changes to the test file.
❌ Features must include changes to the integration test file and the resulting snapshot.
These three changes didn't seem necessary to me.

Thanks.

File: .github/workflows/yarn-upgrade-v1main.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
 
       - name: Locate Yarn cache
         id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
```
Comment (TheRealAmazonKendra) [inline .github/workflows/yarn-upgrade-v1main.yml:29]: No v1 updates on this branch. Only v2 is updated via `main`. In fact, this file should be deleted altogether on `main`.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
I'm not sure that we can necessarily write automated tests for this but we will need documentation of the steps taken for manually testing this before we can accept this change.

**Comment by TheRealAmazonKendra** [comment]
In addition to my comment inline, please make sure that your PR body describes the problem the PR is solving, and the design approach and alternatives considered. Explain why the PR solves the problem. A link to an issue is helpful, but does not replace an explanation of your thought process (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request)).

**Pull request closed without merging**